draft-ietf-tls-wkech.xml

```diff
@@ -482,16 +487,59 @@ Client <-->|                    | |                    | facing
 
 
     <section title="Security Considerations">
+
+        <t>
+            This document defines a way to publish SVCB/HTTPS RR values.
+            If the wrong values were published in the DNS, then TLS clients
+            using ECH might suffer a privacy leak, or degraded service due to
+            overuse of ECH retry_configs.
+        </t>
+
+        <t>
+            When considering the content of SVCB/HTTPS RRs, the general argument for
+            the security of this scheme is that, assuming proper setup and
+            operation of the webPKI, this scheme has the backend server authenticate
+            the JSON structure that is mapped directly to the SVCB/HTTPS RR, to
+            eventually be used by TLS clients when interacting with the backend
+            server, via the client-facing server.
+        </t>
+
+        <t>
+            ECH split-mode security also requires that the backend server
+            acquire SvcParamKey values from the client-facing server via some
+            authenticated means. If the backend server acquires the JSON
+            data authenticated via HTTPS from the client-facing server's public_name
+            then that satisfies this requirement.
+        </t>
+
+        <t>
+            This scheme depends on the webPKI for authentication of entities
+            and results in publication of new SVCB/HTTPS RRs. The webPKI
+            itself however often depends on the DNS to demonstrate control
+            over a DNS name, e.g. when using the ACME protocol
+            <xref target="RFC8555"/> with the HTTP-01 challenge type. A
+            temporary breach of a backend server that allows the attacker to
+            contol the JSON content described here could be used to bootsrap
+            more long-lasting control over the backend's DNS name if the
+            attacker were to request new certificates during the time when
+            the attacker's chosen values were published in the DNS, and if
+            the ACME server doing the validation solely depended on content
+            from the backend's HTTPS RR, e.g. preferring ipv6hints over the
+            AAAA for the backend. It would seem prudent for ACME servers to
+            be cautious if using ipv4hints and ipv6hint, e.g. flagging
+            divergence between those values and A/AAAA RRs.
+        </t>
+
         <t>
-            This document defines another way to publish ECHConfigList values. If the wrong
-            keys were read from here and published in the DNS, then clients
-            using ECH would do the wrong thing, likely resulting in
-            denial of service, or a privacy leak, or worse, when TLS clients attempt to use ECH with
-            a backend web site. So: Don't do that:-)
+            Similarly, a ZF that also has write access to A/AAAA RRs for a
```

Comment:
Similarly to what?  Maybe this paragraph should be moved?

Reply:
moved to be the second paragraph of this section.